include/grpc/event_engine/slice.h

```diff
@@ -0,0 +1,120 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_EVENT_ENGINE_SLICE_H
+#define GRPC_EVENT_ENGINE_SLICE_H
+
+#include <grpc/impl/codegen/port_platform.h>
+
+#include <string>
+#include <utility>
+
+#include <grpc/slice.h>
+
+namespace grpc_event_engine {
+namespace experimental {
+
+class SliceBuffer;
+
+/// A wrapper around \a grpc_slice.
+///
+/// A slice represents a contiguous reference counted array of bytes.
+/// It is cheap to take references to a slice, and it is cheap to create a
+/// slice pointing to a subset of another slice.
+class Slice final {
+  friend class SliceBuffer;
+
+ public:
+  /// Construct an empty slice.
+  Slice() : slice_(grpc_empty_slice()) {}
+  /// Destructor - drops one reference.
+  ~Slice() { grpc_slice_unref(slice_); }
+
+  enum AddRef { ADD_REF };
+  /// Construct a slice from \a slice, adding a reference.
+  Slice(grpc_slice slice, AddRef) : slice_(grpc_slice_ref(slice)) {}
```

Comment:
Let's drop this and the steal ref version, and always steal (i.e. prefer `explicit Slice(grpc_slice slice)`)
It's what's needed most of the time anyway, and the ref counting enums get tricky, and one can always write grpc_slice_ref() if they need a different behavior.